### PR TITLE
Separate flash attention from memory efficient attention.

### PR DESCRIPTION
**Pull request by xinyazhang**
Fixes https://github.com/ROCmSoftwarePlatform/frameworks-internal/issues/5527

### CODE REVIEW

**Comment by xinyazhang** [comment]
Note: testing command `CI=1 python run_test.py --use-pytest -i test_transformers --continue-through-error --exclude-jit-executor --exclude-distributed-tests --verbose`

File: test/test_transformers.py [hunk 1]
```diff
@@ -73,6 +82,15 @@ def get_rtol(true_value: torch.Tensor, computed_value: torch.Tensor) -> float:
         "enable_math": False, "enable_flash": False, "enable_mem_efficient": True}
 }
 
+def get_platform_specific_spda():
```
Comment (jithunnair-amd) [inline test/test_transformers.py:85]: typo: should be "sdpa"?

File: test/test_transformers.py [hunk 2]
```diff
@@ -73,6 +82,15 @@ def get_rtol(true_value: torch.Tensor, computed_value: torch.Tensor) -> float:
...
+    if TEST_CUDA and SM80OrLater:
+        return [SDPBackend.FLASH_ATTENTION, SDPBackend.EFFICIENT_ATTENTION]
+    if TEST_WITH_ROCM:
+        return [SDPBackend.FLASH_ATTENTION]
+    if TEST_CUDA:
+        return [SDPBackend.EFFICIENT_ATTENTION]
```
Comment (jithunnair-amd) [inline test/test_transformers.py:91]: Could be rewritten slightly better to have `TEST_WITH_ROCM` condition first and combine the TEST_CUDA conditions?

File: test/test_transformers.py [hunk 3]
```diff
@@ -10,13 +10,22 @@
 import unittest
 from unittest.mock import patch, MagicMock, ANY
 import math
-from torch.backends.cuda import sdp_kernel, SDPBackend
+# from torch.backends.cuda import sdp_kernel, SDPBackend
```
Comment (jithunnair-amd) [inline test/test_transformers.py:13]: `sdp_kernel` import is commented here, but it is referenced below multiple times?

**Comment by jithunnair-amd** [comment]
Closing as not needed as changes are already in target branch

**Pull request closed without merging**